pkgimage.mk

```diff
@@ -24,129 +24,5 @@ $(JULIA_DEPOT_PATH):
-$(eval $(call stdlib_builder,OpenLibm_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,PCRE2_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,Zlib_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,dSFMT_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,libLLVM_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,libblastrampoline_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,p7zip_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,OpenBLAS_jll,Artifacts Libdl))
-$(eval $(call stdlib_builder,Markdown,Base64))
-$(eval $(call stdlib_builder,Printf,Unicode))
-$(eval $(call stdlib_builder,Random,SHA))
-$(eval $(call stdlib_builder,Logging,StyledStrings))
-$(eval $(call stdlib_builder,Tar,ArgTools,SHA))
-$(eval $(call stdlib_builder,DelimitedFiles,Mmap))
-$(eval $(call stdlib_builder,JuliaSyntaxHighlighting,StyledStrings))
-
-# 2-depth packages
-$(eval $(call stdlib_builder,LLD_jll,Zlib_jll libLLVM_jll Artifacts Libdl))
-$(eval $(call stdlib_builder,LibSSH2_jll,Artifacts Libdl MbedTLS_jll))
-$(eval $(call stdlib_builder,MPFR_jll,Artifacts Libdl GMP_jll))
-$(eval $(call stdlib_builder,LinearAlgebra,Libdl libblastrampoline_jll OpenBLAS_jll))
-$(eval $(call stdlib_builder,Dates,Printf))
-$(eval $(call stdlib_builder,Distributed,Random Serialization Sockets))
-$(eval $(call stdlib_builder,Future,Random))
-$(eval $(call stdlib_builder,UUIDs,Random SHA))
-$(eval $(call stdlib_builder,InteractiveUtils,Markdown))
-
- # 3-depth packages
-$(eval $(call stdlib_builder,LibGit2_jll,MbedTLS_jll LibSSH2_jll Artifacts Libdl))
-$(eval $(call stdlib_builder,LibCURL_jll,LibSSH2_jll nghttp2_jll MbedTLS_jll Zlib_jll Artifacts Libdl))
-$(eval $(call stdlib_builder,REPL,InteractiveUtils Markdown Sockets StyledStrings Unicode))
-$(eval $(call stdlib_builder,SharedArrays,Distributed Mmap Random Serialization))
-$(eval $(call stdlib_builder,TOML,Dates))
-$(eval $(call stdlib_builder,Test,Logging Random Serialization InteractiveUtils))
-
-# 4-depth packages
-$(eval $(call stdlib_builder,LibGit2,LibGit2_jll NetworkOptions Printf SHA Base64))
-$(eval $(call stdlib_builder,LibCURL,LibCURL_jll MozillaCACerts_jll))
-
-# 5-depth packages
-$(eval $(call stdlib_builder,Downloads,ArgTools FileWatching LibCURL NetworkOptions))
-
-# 6-depth packages
-$(eval $(call stdlib_builder,Pkg, Artifacts Dates Downloads FileWatching LibGit2 Libdl\
-								  Logging Markdown Printf REPL Random SHA Serialization\
-								  TOML Tar UUIDs p7zip_jll))
-
-# 7-depth packages
-$(eval $(call stdlib_builder,LazyArtifacts,Artifacts Pkg))
-$(eval $(call stdlib_builder,REPLExt,Pkg REPL))
-
-$(eval $(call stdlib_builder,SparseArrays,Libdl LinearAlgebra Random Serialization SuiteSparse_jll))
-$(eval $(call stdlib_builder,Statistics,LinearAlgebra SparseArrays))
+all-release all-debug:
+	@$(call PRINT_JULIA, $(call spawn,$(JULIA_EXECUTABLE)) --startup-file=no -e 'Base.Precompilation.precompilepkgs(;configs=[``=>Base.CacheFlags(), `--check-bounds=yes`=>Base.CacheFlags(true, 1, 1, true, 2)])')
```

Comment:
Should we make `CacheFlags` `@kwdef` and add a copy constructor for this? So that it is just `CacheFlags(CacheFlags(), check_bounds=2)` (or `=yes`

Reply:
Done